### DISCUSSION TITLE
How to batch messages multiple times?

### DISCUSSION THREAD

**Discussion by webfrank** [redpanda-data/connect#2133] [Q&A]
Hi,
  I have a use case where I collect messages from MQTT, enrich them using a cache and then send out to a web service.
Each message has a "serial" field which I use to enrich and on the http output to group_by and calculate the final message to send.
The http service accepts multiple messages separated by newline and I would like to aggregate the resulting messages further to optimize the process.
How can I re-batch them?

**Comment by mihaitodor** [comment]
> The http service accepts multiple messages separated by newline and I would like to aggregate the resulting messages further to optimize the process.

I'd try to use a `mapping` processor to split that string into an array of substrings like so: `root = content().split("\n").map_each(m -> m.string())` (the `map_each()` part is needed because `content()` returns `bytes` instead of `string` which gets encoded to base64 when serialised as JSON). Then you can add an `unarchive` processor with `format: json_array`. This will produce a batch of messages from the original one. Not sure where the `group_by` you mentioned fits in.
